targets/TARGET_Analog_Devices/mbed_rtx.h

```diff
@@ -0,0 +1,58 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_MBED_RTX_H
+#define MBED_MBED_RTX_H
+
+#if defined(TARGET_EV_COG_AD3029LZ)
+
+#ifndef INITIAL_SP
+#define INITIAL_SP              (0x20004000UL)
+#endif
+#ifndef OS_TASKCNT
+#define OS_TASKCNT              6
+#endif
+#ifndef OS_MAINSTKSIZE
+#define OS_MAINSTKSIZE          112
+#endif
+#ifndef OS_CLOCK
+#define OS_CLOCK                26000000
+#endif
+#ifndef MBED_CONF_APP_THREAD_STACK_SIZE
+#define MBED_CONF_APP_THREAD_STACK_SIZE 1024
```

Comment:
this is app specific config that by default is 4k in mbed, why is this changing it?

Reply:
Was failing one of the tests because the creation of multiple threads were overflowing the heap, hence the limitation of the thread stack size.

Reply:
I believe test should be updated , not target. Target should have default one (all targets are equal).

Reply:
Either that or there should be some guidelines on what minimal stack and heap sizes are required for passing all the tests.

Reply:
> Either that or there should be some guidelines on what minimal stack and heap sizes are required for passing all the tests.

👍  , can you please for now remote this from a target configuration and move this to the test rather, send a separate patch that we can review?

cc @bulislaw @c1728p9 @pan- for visibility of this

Reply:
Removed from mbed_rtx.h. However I can't remember which test I had problem with, will try to find it.